src/constraint_utils.R

```diff
@@ -0,0 +1,47 @@
+# LT 18/05/2022
+
+# utility function for 3D missense constraint metric
+
+## required packages
+# for .PDB structures
+require(bio3d)
+
+# set seed for reproducibility
+set.seed(9011971)
+
+## pairwise distances between residues based on their carbon alpha
+get_dist_mat <- function(pdb_file) {
+  pdb <- read.pdb(pdb_file)
+
+  # calculate distance between residues based on their carbon alpha
+  dist_mat <- dm(pdb, inds = "calpha")
+
+  # dist_mat is upper triangular with empty diagonal: add diagonal, fill lower triangle, convert to matrix
+  dist_mat <- as.matrix(dist_mat)
+  dist_mat[is.na(dist_mat)] <- 0
+  dist_mat <- dist_mat + t(dist_mat)

```

Comment:
Think this simplifies this a little. I don't think you need to specify `as.matrix`, it should still be class `dmat`

```suggestion
  dist_mat <- dm(pdb, inds = "calpha", mask.lower = FALSE)
  dist_mat[is.na(dist_mat)] <- 0
```